pymatgen/analysis/reaction_calculator.py

```diff
@@ -240,7 +240,7 @@ def as_entry(self, energies) -> ComputedEntry:
         Returns a ComputedEntry representation of the reaction.
         """
         relevant_comp = [comp * abs(coeff) for coeff, comp in zip(self._coeffs, self._all_comp)]
-        comp = sum(relevant_comp, Composition())
+        comp: Composition = sum(relevant_comp, Composition())  # type: ignore[assignment]
```

Comment:
Not sure if `sum` could really return `Composition`, please help me double check.

Reply:
yes it can. but `sum` has the wrong type hints in `mypy` restricting it to `int | float`. using `# type: ignore[assignment]` like you did is best

Reply:
Thanks for the input!